### ISSUE TITLE
False positive for "CA2214: Do not call overridable methods in constructors"

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn-analyzers#5286]
### Analyzer

**Diagnostic ID**: [CA2214](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2214): Do not call overridable methods in constructors

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0-preview1.21054.10

### Describe the bug

When the receiver of the virtual call is not `this`, we should assume the instance is fully constructed and there's no problem calling virtual methods on it.

### Steps To Reproduce

```cs
public class MyClass
{
    public MyClass(MyClass other)
    {
        other.M(); // warning
    }

    public virtual void M() { }
}
```

### Expected behavior

No warning on `other.M();

### Actual behavior

Warning on `other.M();`
